### PR TITLE
#feat Add UML Sequence Diagram of successfully authorized API request

### PR DESCRIPTION
**Pull request by vlad-tokarev**
feat Add UML Sequence Diagram of the successfully authorized API request

image

### CODE REVIEW

File: source/gen_security.rst [hunk 1]
```diff
@@ -89,6 +89,11 @@ let's review the API request lifecycle and describe what is happened for each HT
 4. If request if permitted by OpenPolicyAgent_ it is sent to upstream (ODAHU API)
```
Comment (BPylypenko) [inline source/gen_security.rst]:
```suggestion
4. If a request is permitted by OpenPolicyAgent_, it is sent to upstream (ODAHU API)
```

File: source/img/odahu_security_sequence.source [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# this source code can be redered to UML diagram on https://www.websequencediagrams.com/
```
Comment (BPylypenko) [inline source/img/odahu_security_sequence.source]:
```suggestion
# this source code can be rendered to UML diagram on https://www.websequencediagrams.com/
```
I suppose that was meant.

**Review by maxkochubey — APPROVED** [review]

**Comment by maxkochubey** [comment]
Please, replace "NGNIX" to "Nginx" on the picture :)

File: source/img/odahu_security_sequence.source [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# this source code can be rendered to UML diagram on https://www.websequencediagrams.com/
+
+Client -> NGNIX: API request + JWT in cookie
```
Comment (maxkochubey) [inline source/img/odahu_security_sequence.source]:
```suggestion
Client -> NGINX: API request + JWT in cookie
```

**Review by kirillmakhonin — COMMENTED** [review]
Can we replace web based uml generators to something that we can integrate into docs building process? E.g. doT or graphviz or plantuml.

**Comment by vlad-tokarev** [comment]
> Can we replace web based uml generators to something that we can integrate into docs building process? E.g. doT or graphviz or plantuml.

plantuml looks perfect, I haven't found this tool when try to search offline opensource solutions. Thanks. I will try to use it

**Review by BPylypenko — APPROVED** [review]

File: source/img/odahu_security_sequence.source [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+NGNIX -> OAuth2proxy: authenticated?
```
Comment (karbyshevds) [inline source/img/odahu_security_sequence.source]:
```suggestion
NGINX -> OAuth2proxy: authenticated?
```

File: source/img/odahu_security_sequence.source [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+note right of OAuth2proxy: Set Authorization header from cookie
+OAuth2proxy -> NGNIX: Yes
+NGNIX -> envoy sidecar: API request + JWT in header
```
Comment (karbyshevds) [inline source/img/odahu_security_sequence.source]:
```suggestion
NGINX -> envoy sidecar: API request + JWT in header
```

**Review by karbyshevds — CHANGES_REQUESTED** [review]

**Review by dsuslov — APPROVED** [review]

**Review by karbyshevds — APPROVED** [review]

**Review by karbyshevds — APPROVED** [review]

**Review by keshamin — APPROVED** [review]

**Pull request merged**